lib/puppet/indirector/catalog/compiler.rb

```diff
@@ -85,6 +85,14 @@ def add_node_data(node)
 
   # Rewrite a given file resource with the metadata from a fileserver based file
   def replace_metadata(resource, metadata)
+    if resource[:links] == "manage" && metadata.ftype == "link"
+      resource.delete(:source)
```

Comment:
It seems inconsistent to delete `source` here, but not `links => follow` below? Should we delete `source` when `ensure => directory`? Originally we were planning on always preserving `source`, but perhaps it's not needed in some cases? How about if we preserve `source` for now, merge this PR, and talk about when to preserve `source` at today's sync up?